pkg/api/jobs.go

```diff
@@ -147,6 +148,71 @@ func PrintJobsReport(w http.ResponseWriter, req *http.Request, currReport, twoDa
 		limit(req))
 }
 
+// PrintVariantReportFromDB
+func PrintVariantReportFromDB(w http.ResponseWriter, req *http.Request,
+	dbc *db.DB, release string) {
+	// Preferred method of slicing is with start->boundary->end query params in the format ?start=2021-12-02&boundary=2021-12-07.
+	// 'end' can be specified if you wish to view historical reports rather than now, which is assumed if end param is absent.
+	var start time.Time
+	var boundary time.Time
+	var end time.Time
+	var err error
+
+	startParam := req.URL.Query().Get("start")
+	if startParam != "" {
+		start, err = time.Parse("2006-01-02", startParam)
+		if err != nil {
+			RespondWithJSON(http.StatusBadRequest, w, map[string]interface{}{"code": http.StatusBadRequest, "message": fmt.Sprintf("Error decoding start param: %s", err.Error())})
+			return
+		}
+	} else if req.URL.Query().Get("period") == periodTwoDay {
+		// twoDay report period starts 9 days ago, (comparing last 2 days vs previous 7)
+		start = time.Now().Add(-9 * 24 * time.Hour)
+	} else {
+		// Default start to 14 days ago
+		start = time.Now().Add(-14 * 24 * time.Hour)
+	}
+
+	// TODO: currently we're assuming dates use the 00:00:00, is it more logical to add 23:23 for boundary and end? or
+	// for callers to know to specify one day beyond.
```

Comment:
Tough call, my first instinct is for us to assume 23:59. If I make that API call with and end date, I'd expect those results from that day included.

Reply:
This is your comment, FWIW, this is just copy and pasted

Reply:
LOL I am old.

Reply:
It was a good comment :-) Here's an attempt to fix it by adding +24 hours (and move this handling code to it's own function)

https://github.com/openshift/sippy/pull/393/files#diff-f12f97ebaac66a4fd3a647fe92524c5de1bca1f72340d2829891d118516195f7R35